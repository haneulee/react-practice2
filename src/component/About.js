import React, { Component } from "react";

class About extends Component {
  constructor(props) {
    super(props);
    this.state = {
      seconds: Date.now() / 1000
    };
  }
  componentDidMount() {
    this.timerID = setInterval(() => this.tick(), 1000);
  }
  componentWillUnmount() {
    clearInterval(this.timerID);
  }
  tick() {
    this.setState({ seconds: Date.now() / 1000 });
  }
  goHome() {
    if (window.confirm("do you really want to go Home?") === true) {
      this.props.history.push("/");
    }
  }
  render() {
    return (
      <div>
        <h2>About</h2>
        <p>{this.state.seconds} seconds have ellapsed since the UNIX epoch.</p>
        <button className="btn btn-primary" onClick={this.goHome.bind(this)}>
          home
        </button>
      </div>
    );
  }
}

export default About;
