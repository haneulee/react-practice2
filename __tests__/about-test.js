"use strict";

import React from "react";
import About from "../src/component/About";
import renderer from "react-test-renderer";

jest.useFakeTimers();
Date.now = jest.fn(() => 14882363367071);

it("renders correctly", () => {
  const tree = renderer.create(<About />).toJSON();
  expect(tree).toMatchSnapshot();
});
