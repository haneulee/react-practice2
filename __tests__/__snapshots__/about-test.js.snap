// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    About
  </h2>
  <p>
    14882363367.071
     seconds have ellapsed since the UNIX epoch.
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    home
  </button>
</div>
`;
